src/Tools/ExternalAccess/LiveShare/RoslynLSPClientService.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LiveShare;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Client
+{
+    [Export]
+    [ExportCollaborationService(typeof(RoslynLSPClientLifeTimeService),
+                                Scope = SessionScope.Guest,
+                                Role = ServiceRole.LocalService,
+                                Features = "LspServices",
+                                CreationPriority = (int)ServiceRole.LocalService + 2000)]
+    internal class RoslynLSPClientServiceFactory : ICollaborationServiceFactory
+    {
+        private const string RoslynProviderName = "Roslyn";
+        private const string AnyProviderName = "any";
+
+        public ILanguageServerClient ActiveLanguageServerClient { get; private set; }
+
+        public Task<ICollaborationService> CreateServiceAsync(CollaborationSession collaborationSession, CancellationToken cancellationToken)
+        {
+            var languageServerGuestService = (ILanguageServerGuestService)collaborationSession.GetService(typeof(ILanguageServerGuestService));
```

Comment:
what should we do here if we want to separate out type script here.

Reply:
tagging @daytonellwanger

Reply:
TypeScript is already separated out here. The only TypeScript-specific code is lines 51-78. Once you drop TypeScript support, you'd just delete those lines.